coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -58,14 +58,25 @@ class ShardKeyRegexPlanner(val dataset: Dataset,
     val nonMetricShardKeyFilters =
       LogicalPlan.getNonMetricShardKeyFilters(logicalPlan, dataset.options.nonMetricShardColumns)
     if (isMetadataQuery(logicalPlan)
-      || LogicalPlan.hasShardKeyEqualsOnly(logicalPlan, dataset.options.nonMetricShardColumns)) {
+      || (hasRequiredShardKeysPresent(nonMetricShardKeyFilters, dataset.options.nonMetricShardColumns) &&
+      LogicalPlan.hasShardKeyEqualsOnly(logicalPlan, dataset.options.nonMetricShardColumns))) {
       queryPlanner.materialize(logicalPlan, qContext)
     } else if (hasSingleShardKeyMatch(nonMetricShardKeyFilters)) {
       // For queries like topk(2, test{_ws_ = "demo", _ns_ =~ "App-1"}) which have just one matching value
       generateExecWithoutRegex(logicalPlan, nonMetricShardKeyFilters.head, qContext).head
     } else walkLogicalPlanTree(logicalPlan, qContext).plans.head
   }
 
+  def hasRequiredShardKeysPresent(nonMetricShardKeyFilters: Seq[Seq[ColumnFilter]],
+                                  nonMetricShardColumns: Seq[String]): Boolean = {
+    nonMetricShardKeyFilters
+      .foreach(filterGroup => {
+        val columnNames = filterGroup.map(_.column)
+        if (!nonMetricShardColumns.toSet.subsetOf(columnNames.toSet)) return false
+      })
+    true
```

Comment:
Does this always return `true`? Could you add a "should _not_ generate EP for implicit ws query" test?

Reply:
Depends on the query. If query has both nonMetricShardkeys `_ws_` and `_ns_` returns true(e.x `metric{_ws_:foo, _ns_:bar}`) Else false (e.x `metric{ _ns_:bar}`)